src/app/shared/entity/cluster.ts

```diff
@@ -43,18 +77,14 @@ export class Cluster {
   inheritedLabels?: object;
   credential?: string;
 
-  static getProvider(cloud: CloudSpec): string {
-    const providers = Object.keys(cloud);
-    return providers.length > 0 ? providers.pop().toLowerCase() : '';
+  static getProvider(cluster: Cluster): Provider {
+    return Object.values(Provider)
+      .filter(provider => cluster.spec.cloud[provider])
+      .pop();
```

Comment:
This is the fix. It will only pick provider based on the provider enum keys right now.